test-data/unit/check-generics.test

```diff
@@ -513,7 +513,7 @@ Alias[int]("a")  # E: Argument 1 to "Node" has incompatible type "str"; expected
 [out]
 
 [case testTypeApplicationCrash]
-type[int] # this was crashing, see #2302 (comment)  # E: The type "Type[type]" is not generic and not indexable
+type[int] # this was crashing, see #2302 (comment)  # E: "builtins.type" is indexable as a type hint but neither a generic class nor a generic function
```

Comment:
I wouldn't expect this to error at all on 3.9+.

Reply:
I made this message for the following "errors" (taken from the `testBuiltinTypeType` test case):

```python
type[str](1)  # E: "builtins.type" is indexable as a type hint but neither a generic class nor a generic function
r = type[str]("r", (), {})  # E: "builtins.type" is indexable as a type hint but neither a generic class nor a generic function
class S(type[str]): ...  # E: "builtins.type" is indexable as a type hint but neither a generic class nor a generic function
```